diff --git a/src/lib.rs b/src/lib.rs
index 2e65da8..patch001 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -153,7 +153,14 @@ impl MuxAgent {
                     continue;
                 }
             };
-            let agent_identities = client.request_identities().await?;
+            let agent_identities = match client.request_identities().await {
+                Ok(ids) => ids,
+                Err(e) => {
+                    log::warn!(
+                        "Failed to list identities from upstream agent <{}>: {}", sock_path.display(), e);
+                    continue;
+                }
+            };
             {
                 for id in &agent_identities {
                     known_keys.insert(id.pubkey.clone(), sock_path.clone());
